src/oss-detect-cryptography/DetectCryptographyTool.cs

```diff
@@ -92,9 +93,8 @@ private static async Task Main(string[] args)
                             {
                                 targetDirectoryName = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName());
                             }
-                            BaseProjectManager projectManager = ProjectManagerFactory.CreateBaseProjectManager(detectCryptographyTool.HttpClientFactory, targetDirectoryName);
 
-                            string? path = await projectManager.ExtractArchive("temp", File.ReadAllBytes(target));
+                            string? path = await ArchiveHelper.ExtractArchiveAsync(targetDirectoryName, "temp", File.OpenRead(target));
```

Comment:
Nit: As part of the underlying implementation in RecursiveExtractor your provided stream will be used copied to a new Stream inside a FileEntry, because we can't make assumptions about the incoming stream - like that it can be seeked - but doing the extraction requires those properties.

I thought that the Filename based api avoid this extra allocation but it looks like it does not so I have a PR to improve it: https://github.com/microsoft/RecursiveExtractor/pull/77.

You can either call RecursiveExtractor's ExtractToDirectory with a filename or with FileEntry.FromFileName

Reply:
> Some other misc issues.
> 
> 1. Does NugetPackageActions belong in the model directory? It seems like an implementation to me.
> 2. Why does NuGetPackageMetaData not derive from PackageMetadata?

The first one makes sense to me, I can refactor that.
NuGetPackageMetadata doesn't implement PackageMetadata as PackageMetadata was written as a "Catch-All" Metadata model, so it has unnecessary things that don't pertain to NuGet. The NuGetPackageMetadata in this case also doesn't represent what we want as an entire view of a NuGet package's metadata, currently what it is doing is just mapping the `PackageSearchMetadataRegistration` model from `NuGet.Protocol` to a model we control and can expand upon. Such as adding digest and hash data, plus some other data that we as a team have been discussing that we'd like as part of a package's metadata.
We're still flushing out final details, we'll loop you and @scovetta in once we're done to present to you both.